lib/src/git.rs

```diff
@@ -98,6 +99,7 @@ impl GitSettings {
             abandon_unreachable_commits: settings.get_bool("git.abandon-unreachable-commits")?,
             executable_path: settings.get("git.executable-path")?,
             write_change_id_header: settings.get("git.write-change-id-header")?,
+            git_clean_warning: settings.get("git.clean-warning")?,
```

Comment:
This one is unused, and shouldn't be added to `jj-lib` since `clean-warning` is a UI thing.

BTW, I think the original idea was to add `[warnings]` section similar to `[hints]`. I personally don't think we should add warnings which users want to suppress by configuration, though.

Reply:
I still sort of disagree about it being a hint. It most cases where I've encountered hints, hints seem to be emitted after the user has actively done something that may require further action to get the expected result. 

But if the decision is to leave it as a hint, that's fine with me. 

Where should I define the new config option if not in lib? 🤔

Reply:
I just consider that hints are less important notes.

> Where should I define the new config option if not in lib?

add to `cli/src/config/*.toml`.

Reply:
> Being able to silence a warning helps a bad actor hide what they are doing.

Interesting point but I do wonder if this is a real concern for the layer at with `jj` operates.

Reply:
> I just consider that hints are less important notes.
> 
> > Where should I define the new config option if not in lib?
> 
> add to `cli/src/config/*.toml`.

I removed the addition to lib as it was unused and switched this back to a hint.